operations/service_deploy_smoke_utils.go

```diff
@@ -187,12 +187,44 @@ OUTER:
 				return err
 			}
 			page := string(body)
+
+			// Validate that task contains task completed message
 			if strings.Contains(page, "Task completed - SUCCESS") {
 				grip.Infof("Found task completed message in log:\n%s", page)
 			} else {
 				grip.Errorf("did not find task completed message in log:\n%s", page)
 				return errors.New("did not find task completed message in log")
 			}
+
+			// Validate that setup_group only runs in first task
+			if strings.Contains(page, "first") {
+				if !strings.Contains(page, "setup_group") {
+					return errors.New("did not find setup_group in logs for first task")
+				}
+			} else {
+				if strings.Contains(page, "setup_group") {
+					return errors.New("setup_group should only run in first task")
+				}
+			}
+
+			// Validate that setup_task and teardown_task run for all tasks
+			if !strings.Contains(page, "setup_task") {
+				return errors.New("did not find setup_task in logs")
+			}
+			if !strings.Contains(page, "teardown_task") {
```

Comment:
should these be constants somewhere (probably in evergreen)

Reply:
I don't think so. The other place they are used is in struct tags.